src/Core/Managed/Shared/Extensibility/Implementation/SimpleMetricStatisticsAggregator.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents mechanism to calculate basic statistical parameters of a series of numeric values.
+    /// </summary>
+    internal class SimpleMetricStatisticsAggregator
+    {
```

Comment:
So having read the other bits, the only way this can be a problem is when Track was called right before a Snapshot is made and then lost the race. We would expose the spinlock internally and lock the aggregator while making the snapshot. Because the race is very rare, the lock will be contended very rarely and so should not result in any significant perf decrease.